codejail/tests/helpers.py

```diff
@@ -3,28 +3,85 @@
 """
 
 from contextlib import contextmanager
+import os.path
+import sys
+from unittest import SkipTest
+
+from codejail import jail
+from codejail import languages
 
-from codejail import jail_code
 
 SAME = object()
 
 
 @contextmanager
-def override_configuration(command, bin_path, user):
+def override_configuration(command, bin_path=SAME, user=SAME, lang=SAME):
     """
     Context manager to temporarily alter the configuration of a codejail
     command.
     """
-    old = jail_code.COMMANDS.get(command)
+    old = jail.COMMANDS.get(command)
     if bin_path is SAME:
-        bin_path = old['cmdline_start'][0]
+        bin_path = old.bin_path
     if user is SAME:
-        user = old['user']
+        user = old.user
+    if lang is SAME:
+        lang = old.lang
     try:
-        jail_code.configure(command, bin_path, user)
+        jail.configure(command, bin_path, user, lang)
         yield
     finally:
         if old is None:
-            del jail_code.COMMANDS[command]
+            del jail.COMMANDS[command]
+        else:
+            jail.COMMANDS[command] = old
+
+
+class JailMixin(object):
+    """
+    Mixin to add a default "python" jail environment.
+
+    Can be configured by specifying `CODEJAIL_TEST_VENV` and
+    `CODEJAIL_TEST_USER` environment variables.  Defaults to the path of the
+    current virtualenv with -sandbox appended and the user named `'sandbox'`.
+    """
+
+    _codejail_venv = os.environ.get('CODEJAIL_TEST_VENV')
+    _codejail_user = os.environ.get('CODEJAIL_TEST_USER', 'sandbox')
+
+    def setUp(self):
+        super(JailMixin, self).setUp()
+        print '1', self._codejail_venv
+        if not jail.is_configured("python"):
+            if not self._codejail_venv:
+                self._codejail_venv = self._autoconfigure_codejail_venv()
+            if not self._codejail_user:
+                # User explicitly requested no su user via environment variable
+                self._codejail_user = None
+        bin_path = os.path.join(self._codejail_venv, 'bin/python2')
+        jail.configure("python", bin_path, user=self._codejail_user, lang=languages.python2)
+
+    def _autoconfigure_codejail_venv(self):
+        """
+        For the purposes of tests, look for a sandbox alongside the currently
+        running python.
+        """
+        codejail_venv = '{}-sandbox'.format(sys.prefix)
+        if os.path.isdir(codejail_venv):
+            return codejail_venv
         else:
-            jail_code.COMMANDS[command] = old
+            raise SkipTest("No virtualenv found for codejail")
```

Comment:
I forget if this is how it was before.  Skipping tests because we don't have the prerequisites to run them can be dangerous.  Much of the stuff in the code for running without real apparmor, etc, was because devs at the time didn't all use devstack, and expecting them to configure apparmor was too much.  Now that people use devstack, and apparmor is pre-configured, maybe we can be stricter.